sources/views/form.js

```diff
@@ -7,83 +7,88 @@ import { contacts } from "../models/contacts";
-                                } else if (form.isDirty()) {
-                                    form.setDirty();
-                                        contacts.updateItem(formData.id, formData);                                 
-                                    webix.message("Contact is updated.");
-                                } else {
-                                    webix.message("Contact hasn't been changed.");
-                                }
-                            }
-                        },
-                        {
-                            view: "button",
-                            value: _("Cancel"),
-                            click: () => {
-                                const form = this.getRoot();
-                                if (form.isDirty()) {
-                                    this._jetPopup.showWindow();
-                                } else {
-                                    webix.message("The form has not been changed. Nothing to reset.");
-                                }
+        return webix.promise.all([
+            contacts.waitData,
+            statuses.waitData,
+            countries.waitData
+        ]).then(() => {
+            return {
+                view: "form",
+                localId: "contactForm",
+                margin: 20,
+                elements: [
+                    {
+                        view: "text",
+                        type: "text",
+                        label: "Name",
+                        name: "Name"
+                    },
+                    {
+                        view: "text",
+                        type: "email",
+                        label: "Email",
+                        name: "Email"
+                    },
+                    {
+                        view: "combo",
+                        label: "Status",
+                        name: "Status",
+                        options: {
+                            body: {
+                                template: "#Name#"
                             },
+                            data: statuses
                         }
-                    ]
-                },
-                {}
-            ], css: "webix_shadow_medium app_start"
-        };
-
+                    },
+                    {
+                        view: "combo",
+                        label: "Country",
+                        name: "Country",
+                        options: {
+                            body: {
+                                template: "#Name#"
+                            },
+                            data: countries
+                        }
+                    },
+                    {
+                        cols: [
+                            {
+                                view: "button",
+                                value: _("Save"),
+                                css: "webix_primary",
+                                click: () => {
```

Comment:
now it's time for some code style perfectionism
the code of config() got long, so you should move handlers to class methods. e.g. click: () => this.SaveForm(), where SaveForm() contains the code of this click handler

Reply:
mb this is also true for other view classes